imap_processing/codice/packet_definitions/P_COD_HI_INSTRUMENT_COUNTERS.xml

```diff
@@ -39,14 +39,9 @@
 			<xtce:IntegerParameterType name="UINT32" signed="false">
 				<xtce:IntegerDataEncoding sizeInBits="32" encoding="unsigned" />
 			</xtce:IntegerParameterType>
-			<xtce:BinaryParameterType name="BYTE4325376">
```

Comment:
Why did this change? telemetry generator should do this `BinaryParameterType`, and not the integer type which it changed to.

Reply:
Hmmm...I can look into it, I didn't touch anything except the typo issue. The `def create_parameter_types` didn't change, so I honestly do not know currently.

Reply:
Ok! So.. @maxinelasp @greglucas the code is correct on my local, but I figured out thet Maxine added `or "INT"` when she updated the `telemetry_generator.py` in line 118. 
I simply took this away, and codice returned to `BinaryParameterType` like before. Solutions?

Reply:
I am going to leave it with Maxines current update. Since this is only codice related right now..we will handle this error when we run into it.

Reply:
Are you working from the `TLM_COD.xlsx` spreadsheet that is in the repo? I see BYTE in the dataType field under the `P_COD_HI_INSTRUMENT_COUNTERS` tab, so I think that should still fall down to BinaryParameterType.

p.s. if you click on the line number, then expand the three dots you can copy permalink to get a reference to the code spot you're talking about.
<img width="752" alt="image" src="https://github.com/IMAP-Science-Operations-Center/imap_processing/assets/12417828/afb596e1-b256-4637-b19f-34f0fb898b40">

https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/01d9a0df54206c7fc907e511232f0a41aef56611/tools/xtce_generation/telemetry_generator.py#L118

And view the git blame directly in the UI if that is helpful as well.

Reply:
I agree with Greg, if it says "BYTE" it should be in the "BYTE" path, resulting in a BinaryParameterType. What does the spreadsheet say? If you'd like to leave it and come back to it, then I guess that's ok since this is a bug we want fixed. But, I'm concerned about this and I will double check my GLOWS XTCE files when I rerun this code to make sure the BinaryParameterType is correct there.

Reply:
OK, I think I see now that I look closer at the code.

```python
"SINT" or "INT" in parameter_type_ref_name
```
will evaluate to "SINT" if "INT" not in parameter_type_ref_name. Which then in turn evaluates to True, and thus we never hit the BYTE path.

I think it might need to be something more like this:
```python
any(x in parameter_type_ref_name for x in ["SINT", "INT"])
```

Reply:
I can fix it in this PR, but would need guidance. I kinda see what you mean here Greg, but I also haven't built many `any` statements with `for` loops.

Reply:
Wait! I think I got it!

Reply:
Actually, it might just be easier to change the order or lookup around. What is `parameter_type_ref_name` is it just a string, or is it a tuple/set/list?

A few options for you:
```python
elif "INT" in parameter_type_ref_name:
```
or
```python
elif parameter_type_ref_name in ["SINT", "INT"]:
```

Reply:
I just made a new commit! It seemed to fix this issue.